src/index.tsx

```diff
@@ -271,6 +280,45 @@ class PyflyByWidget extends Widget {
     }
   }
 
+  async sendTidyImportRequest(): Promise<any> {
+    const cells = this._context.model.cells;
+    const cellArray = [];
+    for (let i = 0; i < cells.length; ++i) {
+      cellArray.push({
+        code: cells.get(i).value.text,
+        type: cells.get(i).type
+      });
+    }
+    const comm = this._comms[PYFLYBY_COMMS.TIDY_IMPORTS];
+    if (comm && !comm.isDisposed) {
+      comm.send({
+        type: PYFLYBY_COMMS.TIDY_IMPORTS,
+        cellArray: cellArray
+      });
+    }
+  }
+
+  refillCells(cellArray: any, imports: any): void {
+    const cells = this._context.model.cells;
+    for (let i = 0; i < cellArray.length; ++i) {
```

Comment:
While I'm not 100% sure, but I think we should guard against race cases here. E.g. a user pushes the button then edits text, it should refuse to replace.

Reply:
I'm still thinking about a solution for this. Lab doesn't provide an API to block such operations so my first thought here is to show a spinner in the notebook panel while tidy-imports is running. Thoughts?

Reply:
I was thinking of using an ID for all messages and checking that it sends the same one back?

Reply:
I've added a checksum that is passed along with the code, if checksum is same before and after receiving response, we modify the notebook, else we tell the user that TidyImports couldn't be run.